assets/src/api/use-get-supervisors.tsx

```diff
@@ -1,5 +1,24 @@
 import { $api } from '~/shared/api';
 import { API_ENDPOINTS } from '~/shared/const';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { toastManager } from '~/shared/lib/toastManager';
 
-export const getSupervisors = (): Promise<ApiSupervisor[]> =>
-  $api.get<ApiSupervisor[]>(API_ENDPOINTS.SUPERVISORS()).then(response => response.data);
+export const useGetSupervisors = () =>
+  useQuery({
+    queryKey: ['supervisors'],
+    queryFn: () => $api.get<ApiSupervisor[]>(API_ENDPOINTS.SUPERVISORS()),
+  });
+
+export const useInvalidateSupervisors = () => {
+  const queryClient = useQueryClient();
+
+  const revalidatingNotification = (promise: Promise<void>) => {
+    toastManager.promise(promise, {
+      loading: 'Revalidating...',
```

Comment:
What does Revalidate mean? Will this be shown to the end user?

Reply:
Resolved to "Updating servers data..."